cybozu/reactor.hpp

```diff
@@ -202,7 +261,7 @@ class reactor {
     // calling this.
```

Comment:
ditto

Reply:
What do you mean?
`request_removal` should be used non-reactor threads, and there's already such a comment.

Reply:
The general lock rules for `m_fd` are as follows

* No lock is required when reading `m_fd` from the reactor thread
* Read lock is required when reading `m_fd` from a non-reactor thread

However, this function does not take a read lock even though it is executed in a non-reactor thread. I think there should be a comment as to why this is safe (i.e., no new operation is performed on the `m_fd` that called `request_removal`).

Reply:
OK.
I'll add the following comment.
```
    // Note that the use of `m_fd` is safe here because the file descriptor
    // is guaranteed to be alive at this point. It will be closed only after
    // the resource put in `m_drop_req` is processed by the reactor thread.
```

Reply:
done

Reply:
I don't think the fact that `m_fd` is alive is a reason to omit read lock.
In order to omit read lock in a non-reactor thread, it is important to state that the read does not create a new operation on the fd. This is because if you can get a write lock on `try_close`, it must be guaranteed that there is no operation in progress for that fd.

Reply:
To be honest, I don't think it's worth adding comments here because
it is very clear to me that the function does not initiate any syscalls
with `m_fd` here.

In fact, `m_fd` is used as a map key and this usage is safe because
`m_fd` is a constant.

Reply:
This may be obvious to those with comprehensive knowledge of the program. But to me (and newbies) it does not seem obvious. The code stores the fd in a data structure, and it is possible to read this fd from anywhere in the reactor. It takes time to make sure that no new operations are initiated for this fd.

Reply:
OK. will update the comment.

Reply:
updated